include/seqan3/io/detail/misc_input.hpp

```diff
@@ -102,7 +101,7 @@ inline auto make_secondary_istream(std::basic_istream<char_t> & primary_stream,
     std::string const extension = filename.extension().string();
 
     // set return value appropriately
-    if (read_chars == magic_number.size() && contrib::_bgzfCheckHeader(magic_number.data())) // BGZF
+    if (read_chars == magic_number.size() && contrib::_bgzfCheckHeader(magic_number.data()) && extension != ".bam") // BGZF
```

Comment:
length

Mhm, I don't know whats supposedly faster, but I would put the fastest condition at the front.

Reply:
I think the size check at the front is already the fastest? The others compare characters linearly.

If you were aiming at "which condition is most likely false", since this is terminating the other comparisons it is also the size I guess. the extension != ".bam" is the most likely one as this function is also used for other formats.